service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/NoticeOfChangeControllerSubmittedTest.java

```diff
@@ -0,0 +1,114 @@
+package uk.gov.hmcts.reform.fpl.controllers;
+
+import org.junit.jupiter.api.Test;
+import org.springframework.boot.test.autoconfigure.OverrideAutoConfiguration;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import uk.gov.hmcts.reform.ccd.model.Organisation;
+import uk.gov.hmcts.reform.fpl.model.CaseData;
+import uk.gov.hmcts.reform.fpl.model.Respondent;
+import uk.gov.hmcts.reform.fpl.model.RespondentSolicitor;
+import uk.gov.hmcts.reform.fpl.model.notify.noticeofchange.NoticeOfChangeRespondentSolicitorTemplate;
+import uk.gov.service.notify.NotificationClient;
+
+import java.util.Map;
+
+import static org.mockito.Mockito.verify;
+import static uk.gov.hmcts.reform.fpl.NotifyTemplates.NOTICE_OF_CHANGE_FORMER_REPRESENTATIVE;
+import static uk.gov.hmcts.reform.fpl.NotifyTemplates.NOTICE_OF_CHANGE_NEW_REPRESENTATIVE;
+import static uk.gov.hmcts.reform.fpl.utils.AssertionHelper.checkUntil;
+import static uk.gov.hmcts.reform.fpl.utils.ElementUtils.wrapElements;
+
+@WebMvcTest(NoticeOfChangeController.class)
+@OverrideAutoConfiguration(enabled = true)
+class NoticeOfChangeControllerSubmittedTest extends AbstractCallbackTest {
+    private static final Long CASE_ID = 10L;
+    private static final String CASE_NAME = "Test";
+    private static final String SOLICITOR_EMAIL = "solicitor@email.com";
```

Comment:
The same `SOLICITOR_EMAIL` is being used to test the 'access given' and 'access revoked' notifications for the new and old solicitors respectively - but if both were being sent to the same solicitor it wouldn't fail. 

This would be a stronger test if separate email addresses are used for each solicitor.